imap_processing/ultra/l1b/ultra_l1b_extended.py

```diff
@@ -315,3 +319,96 @@ def get_ssd_back_position_and_tof_offset(
+    back to the coincidence anode.
+
+    Further description is available on pages 34-35 of
+    IMAP-Ultra Flight Software Specification document
+    (7523-9009_Rev_-.pdf).
+
+    Parameters
+    ----------
+    de_dataset : xarray.Dataset
+        Data in xarray format.
+    particle_tof : np.ndarray
+        Particle time of flight (i.e. from start to stop)
+        (tenths of a nanosecond).
+    sensor : str
+        Sensor name.
+
+    Returns
+    -------
+    etof : np.ndarray
+        Time for the electrons to travel back to
+        coincidence anode (tenths of a nanosecond).
+    xc : np.ndarray
+        X coincidence position (hundredths of a millimeter).
+    """
+    index_top = np.nonzero(np.isin(de_dataset["COIN_TYPE"], CoinType.Top.value))[0]
+    de_top = de_dataset.isel(epoch=index_top)
+
+    index_bottom = np.nonzero(np.isin(de_dataset["COIN_TYPE"], CoinType.Bottom.value))[
+        0
+    ]
+    de_bottom = de_dataset.isel(epoch=index_bottom)
+
+    t1 = np.zeros(len(de_dataset["COIN_TYPE"]), dtype=np.float64)
+    t2 = np.zeros(len(de_dataset["COIN_TYPE"]), dtype=np.float64)
+
+    xc = np.zeros(len(de_dataset["COIN_TYPE"]), dtype=np.float64)
+    etof = np.zeros(len(de_dataset["COIN_TYPE"]), dtype=np.float64)
+
+    # Normalized TDCs
+    # For the stop anode, there are mismatches between the coincidence TDCs,
+    # i.e., CoinN and CoinS. They must be normalized via lookup tables.
+
+    # TpCoinNNorm Top
+    coin_n_norm_top = get_norm(de_top["COIN_NORTH_TDC"], "CoinN", sensor)
+    # TpCoinSNorm Top
+    coin_s_norm_top = get_norm(de_top["COIN_SOUTH_TDC"], "CoinS", sensor)
+
+    t1[index_top] = coin_n_norm_top + coin_s_norm_top  # /2 incorporated into scale
+    xc[index_top] = get_image_params("XCOINTPSC") * (
+        coin_s_norm_top - coin_n_norm_top
+    ) + get_image_params("XCOINTPOFF")  # millimeter
+    # Time for the electrons to travel back to coincidence anode.
+    t2[index_top] = get_image_params("ETOFSC") * t1[index_top] + get_image_params(
+        "ETOFTPOFF"
+    )
+
+    # Multiply by 10 to convert to tenths of a nanosecond.
+    etof[index_top] = t2[index_top] * 10 - particle_tof[index_top]
+
+    # TpCoinNNorm Bottom
+    coin_n_norm_bottom = get_norm(de_bottom["COIN_NORTH_TDC"], "CoinN", sensor)
+    # TpCoinSNorm Bottom
+    coin_s_norm_bottom = get_norm(de_bottom["COIN_SOUTH_TDC"], "CoinS", sensor)
+    t1[index_bottom] = (
+        coin_n_norm_bottom + coin_s_norm_bottom
+    )  # /2 incorporated into scale
+    xc[index_bottom] = get_image_params("XCOINBTSC") * (
+        coin_s_norm_bottom - coin_n_norm_bottom
+    ) + get_image_params("XCOINBTOFF")  # millimeter
+
+    # Time for the electrons to travel back to coincidence anode.
+    t2[index_bottom] = get_image_params("ETOFSC") * t1[index_bottom] + get_image_params(
+        "ETOFBTOFF"
+    )
+
+    # Multiply by 10 to convert to tenths of a nanosecond.
+    etof[index_bottom] = t2[index_bottom] * 10 - particle_tof[index_bottom]
```

Comment:
This code section is essentially a duplicate of lines 377-392. Consider extracting into a function.